{"priority": "minor", "kind": "bug", "repository": {"links": {"self": {"href": "data/repositories/dedalus-project/dedalus.json"}, "html": {"href": "#!/dedalus-project/dedalus"}, "avatar": {"href": "data/bytebucket.org/ravatar/{1dc39ab6-2798-450d-af2f-e976f94b5794}ts=2009435"}}, "type": "repository", "name": "dedalus", "full_name": "dedalus-project/dedalus", "uuid": "{1dc39ab6-2798-450d-af2f-e976f94b5794}"}, "links": {"attachments": {"href": "data/repositories/dedalus-project/dedalus/issues/68/attachments_page=1.json"}, "self": {"href": "data/repositories/dedalus-project/dedalus/issues/68.json"}, "watch": {"href": "https://api.bitbucket.org/2.0/repositories/dedalus-project/dedalus/issues/68/watch"}, "comments": {"href": "data/repositories/dedalus-project/dedalus/issues/68/comments_page=1.json"}, "html": {"href": "#!/dedalus-project/dedalus/issues/68/bugs-in-2d_rayleigh_benard-example-script"}, "vote": {"href": "https://api.bitbucket.org/2.0/repositories/dedalus-project/dedalus/issues/68/vote"}}, "reporter": {"display_name": "Mikhail Schee", "uuid": "{34cab4bf-72fd-4786-8e5b-c1cd0a1cdafb}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7B34cab4bf-72fd-4786-8e5b-c1cd0a1cdafb%7D"}, "html": {"href": "https://bitbucket.org/%7B34cab4bf-72fd-4786-8e5b-c1cd0a1cdafb%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/1b1e91cf6eff3334f80c736291f17c1fd=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsMS-5.png"}}, "nickname": "Mikhail Schee", "type": "user", "account_id": "5d8a65a54f81660dca8d12fd"}, "title": "Bugs in 2d_rayleigh_benard example script", "component": null, "votes": 0, "watches": 1, "content": {"raw": "When downloading Dedalus, there is a script in examples/ivp/2d\\_rayleigh\\_benard called \u201crayleigh\\_benard.py\u201d which has a few bugs. I downloaded and installed Dedalus version 2.1810 on 2019/09/23.\r\n\r\n\u200c\r\n\r\nThe first issue is in line 134 which reads \u201cwhile solver.proceed:\u201d which causes this error:\r\n\r\n\u200c\r\n\r\nAttributeError: 'InitialValueSolver' object has no attribute 'proceed'\r\n\r\n\u200c\r\n\r\nI fixed this issue by changing line 134 to read \u201cwhile solver.ok:\u201d\r\n\r\n\u200c\r\n\r\nThe second issue is that then the solver only ran to 10 iterations instead of to the specified stopping simulation time. I fixed this by adding two lines which explicitly set the stopping wall time and iterations to be infinite:\r\n\r\n\u200c\r\n\r\nsolver.stop\\_wall\\_time = np.inf\r\n\r\nsolver.stop\\_iteration = np.inf\r\n\r\n\u200c", "markup": "markdown", "html": "<p>When downloading Dedalus, there is a script in examples/ivp/2d_rayleigh_benard called \u201crayleigh_benard.py\u201d which has a few bugs. I downloaded and installed Dedalus version 2.1810 on 2019/09/23.</p>\n<p>\u200c</p>\n<p>The first issue is in line 134 which reads \u201cwhile solver.proceed:\u201d which causes this error:</p>\n<p>\u200c</p>\n<p>AttributeError: 'InitialValueSolver' object has no attribute 'proceed'</p>\n<p>\u200c</p>\n<p>I fixed this issue by changing line 134 to read \u201cwhile solver.ok:\u201d</p>\n<p>\u200c</p>\n<p>The second issue is that then the solver only ran to 10 iterations instead of to the specified stopping simulation time. I fixed this by adding two lines which explicitly set the stopping wall time and iterations to be infinite:</p>\n<p>\u200c</p>\n<p>solver.stop_wall_time = np.inf</p>\n<p>solver.stop_iteration = np.inf</p>\n<p>\u200c</p>", "type": "rendered"}, "assignee": null, "state": "invalid", "version": null, "edited_on": null, "created_on": "2019-09-24T18:58:52.757225+00:00", "milestone": null, "updated_on": "2019-09-24T20:35:46.707974+00:00", "type": "issue", "id": 68}